src/vs/workbench/electron-browser/integration.ts

```diff
@@ -134,12 +134,24 @@ export class ElectronIntegration {
 		ipc.on('vscode:enterFullScreen', (event) => {
 			this.partService.joinCreation().then(() => {
 				this.partService.addClass('fullscreen');
+				// We need to re-layout the sidebar as the activitybar's width can change between
+				// fullscreen and window'd modes on macOS.
+				this.partService.setSideBarHidden(this.partService.isSideBarHidden());
+
 			});
 		});
 
 		ipc.on('vscode:leaveFullScreen', (event) => {
 			this.partService.joinCreation().then(() => {
 				this.partService.removeClass('fullscreen');
+				this.partService.setSideBarHidden(this.partService.isSideBarHidden());
```

Comment:
this is ugly, we should rather call `this.partService.layout()` and make it work there if it does not